pkg/service/model_build_target_group.go

```diff
@@ -0,0 +1,312 @@
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCTimeout, &timeoutSeconds, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return timeoutSeconds, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckHealthyThresholdCount(_ context.Context) (int64, error) {
+	healthyThresholdCount := t.defaultHealthCheckHealthyThreshold
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCHealthyThreshold, &healthyThresholdCount, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return healthyThresholdCount, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckUnhealthyThresholdCount(_ context.Context) (int64, error) {
+	unhealthyThresholdCount := t.defaultHealthCheckUnhealthyThreshold
+	if _, err := t.annotationParser.ParseInt64Annotation(annotations.SvcLBSuffixHCUnhealthyThreshold, &unhealthyThresholdCount, t.service.Annotations); err != nil {
+		return 0, err
+	}
+	return unhealthyThresholdCount, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetHealthCheck(ctx context.Context) (*elbv2model.TargetGroupHealthCheckConfig, error) {
+	healthCheckProtocol, err := t.buildTargetGroupHealthCheckProtocol(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var healthCheckPathPtr *string
+	if healthCheckProtocol != elbv2model.ProtocolTCP {
+		healthCheckPathPtr = t.buildTargetGroupHealthCheckPath(ctx)
+	}
+	healthCheckPort, err := t.buildTargetGroupHealthCheckPort(ctx)
+	if err != nil {
+		return nil, err
+	}
+	intervalSeconds, err := t.buildTargetGroupHealthCheckIntervalSeconds(ctx)
+	if err != nil {
+		return nil, err
+	}
+	timeoutSeconds, err := t.buildTargetGroupHealthCheckTimeoutSeconds(ctx)
+	if err != nil {
+		return nil, err
+	}
+	healthyThresholdCount, err := t.buildTargetGroupHealthCheckHealthyThresholdCount(ctx)
+	if err != nil {
+		return nil, err
+	}
+	unhealthyThresholdCount, err := t.buildTargetGroupHealthCheckUnhealthyThresholdCount(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &elbv2model.TargetGroupHealthCheckConfig{
+		Port:                    &healthCheckPort,
+		Protocol:                &healthCheckProtocol,
+		Path:                    healthCheckPathPtr,
+		IntervalSeconds:         &intervalSeconds,
+		TimeoutSeconds:          &timeoutSeconds,
+		HealthyThresholdCount:   &healthyThresholdCount,
+		UnhealthyThresholdCount: &unhealthyThresholdCount,
+	}, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupAttributes(_ context.Context) ([]elbv2model.TargetGroupAttribute, error) {
+	var rawAttributes map[string]string
+	if _, err := t.annotationParser.ParseStringMapAnnotation(annotations.SvcLBSuffixTargetGroupAttributes, &rawAttributes, t.service.Annotations); err != nil {
+		return nil, err
+	}
+	if rawAttributes == nil {
+		rawAttributes = make(map[string]string)
+	}
+	if _, ok := rawAttributes[TGAttrsProxyProtocolV2Enabled]; !ok {
+		rawAttributes[TGAttrsProxyProtocolV2Enabled] = "false"
```

Comment:
why do we want to set it to false here. i think we can leave it unset, so AWS decides the default. or have a defaultProxyProtocolV2Enabled variable in modelBuildTask

Reply:
With the aws-load-balancer-proxy-protocol annotation, we need to make sure we disable this if the annotation gets deleted or not specified. I will add the defaultProxyProtocolV2Enabled in the modelBuildTask.